susemanager/bin/pg-migrate-x-to-y.sh

```diff
@@ -74,101 +92,113 @@ else
         FREESPACE=`expr $FREESPACE / 1024 / 1024`
 
         echo
-        echo "`date +"%H:%M:%S"`   Insufficient diskspace in $DIR ($NEEDSPACE GB needed, $FREESPACE GB available)!"
+        echo "$(timestamp)   Insufficient diskspace in $DIR ($NEEDSPACE GB needed, $FREESPACE GB available)!"
         echo
-        echo "`date +"%H:%M:%S"`   A fast migration does not need this additional diskspace, because"
-        echo "`date +"%H:%M:%S"`   database files will be hardlinked instead of copied."
+        echo "$(timestamp)   A fast migration does not need this additional diskspace, because"
+        echo "$(timestamp)   database files will be hardlinked instead of copied."
         echo
-        echo "`date +"%H:%M:%S"`   If you have a backup of the database, run \"$0 fast\""
+        echo "$(timestamp)   If you have a backup of the database, run \"$0 fast\""
         echo
         exit 1
     else
         echo "ok."
     fi
 fi
 
-echo "`date +"%H:%M:%S"`   Shut down spacewalk services..."
+echo "$(timestamp)   Shut down spacewalk services..."
 spacewalk-service stop
+if [ ${?} -ne 0 ]; then
+    echo "$(timestamp)   At least one spacewalk service failed to stop!"
+    echo "$(timestamp)   Please check the logs at /var/log/rhn and /var/log/tomcat"
+    exit 1
+fi
+
 systemctl stop postgresql
+if [ ${?} -ne 0 ]; then
+    echo "$(timestamp)   PostgresSQL failed to stop!"
+    echo "$(timestamp)   Please check systemd logs for PostreSQL"
+    exit 1
+fi
 
-echo "`date +"%H:%M:%S"`   Checking postgresql version..."
+echo "$(timestamp)   Checking postgresql version..."
 rpm -q postgresql$NEW_VERSION-server > /dev/null 2>&1
 if [ $? -eq 0 ]; then
-    echo "`date +"%H:%M:%S"`   postgresql $NEW_VERSION is already installed. Good."
+    echo "$(timestamp)   postgresql $NEW_VERSION is already installed. Good."
 else
-    echo "`date +"%H:%M:%S"`   Installing postgresql $NEW_VERSION..."
+    echo "$(timestamp)   Installing postgresql $NEW_VERSION..."
     zypper --non-interactive in postgresql$NEW_VERSION postgresql$NEW_VERSION-contrib postgresql$NEW_VERSION-server
 
     if [ ! $? -eq 0 ]; then
```

Comment:
Now that I see this, it's a strange way of thing things, negating the whole condition instead of using `-ne` :-D